blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BlueQueueContainer.java

```diff
@@ -1,7 +1,46 @@
 package io.jenkins.blueocean.rest.model;
 
+import org.kohsuke.stapler.WebMethod;
+import org.kohsuke.stapler.interceptor.JsonOutputFilter;
+import org.kohsuke.stapler.json.JsonResponse;
+import org.kohsuke.stapler.verb.POST;
+import org.kohsuke.stapler.verb.PUT;
+
 /**
  * @author Ivan Meredith
  */
 public abstract class BlueQueueContainer extends Container<BlueQueueItem> {
+    @WebMethod(name = "new")
+    @JsonResponse
+    @JsonOutputFilter(excludes = "state")
+    @POST
+    public abstract BlueQueueItemCreateResponse createItem();
+
+    /**
+     * Response object for create item. is a mirror of BlueQueueItem.
+     */
+    public static class BlueQueueItemCreateResponse {
+        private BlueQueueItem item;
+
+        public BlueQueueItemCreateResponse(BlueQueueItem item) {
+            this.item = item;
+
+        }
+
+        public String getId() {
+            return item.getId();
+        }
+
+        public String getPipeline() {
+            return item.getPipeline();
+        }
+
+        public String getQueuedTime() {
+            return item.getQueuedTimeString();
+        }
+
+        public int getExpectedBuildNumber() {
+            return item.getExpectedBuildNumber();
+        }
```

Comment:
Why do we return this? The `Run` instance now (since middle/late 2015) has the queueId on it for the queue item that caused the `Run`, allowing you to reliably figure out the corresponding `Run`. There's also the SSE events for the client side. So ... seeing as this seems a bit dubious (what happens if the "unexpected" happens?), then I'm not sure I get why would expose it.